### PR TITLE
add css lint task

### PR DESCRIPTION
**Pull request by tsm91**
As discussed here #556. This is the css lint task implementation. I messed up and created #561 from the master branch, so this includes all of #561's changes. That one should be merged / rejected first.

linters:
* stylelint
* doiuse
* colorguard

### CODE REVIEW

File: gulpfile.ts [hunk 1]
```diff
@@ -43,8 +45,10 @@ gulp.task('build.e2e', done =>
 gulp.task('build.prod', done =>
   runSequence('clean.prod',
               'tslint',
+              'css-lint',
```
Comment (mgechev) [inline gulpfile.ts]: Would you name the file with `_` instead if dash or better remove the dash at all, in order to follow the convention in the rest of the tasks.

**Comment by mgechev** [comment]
LGTM, great job!

Do you think we can make csslint make sure the project is following [Google's style guide](https://google.github.io/styleguide/htmlcssguide.xml)?

**Comment by mgechev** [comment]
https://github.com/mgechev/angular2-seed/pull/561 seems part of this PR. Lets close it.

File: gulpfile.ts (hunk 1 above)
Comment (tsm91) [inline gulpfile.ts]: done

**Comment by mgechev** [comment]
@tsm91 what do you think about the Google style guide? Are there any ready-to-use configuration files? If not, we can stick to the current rules we already have.

**Comment by tsm91** [comment]
> Do you think we can make csslint make sure the project is following Google's style guide?

I think so. Would make sense to use the google css guidelines. In fact the [stylelint-config-standard](https://github.com/stylelint/stylelint-config-standard#stylelint-config-standard) uses some part of Google's style guides. 

Unfortunately i haven't found a stylelint rule implementation of it. I am sure it is possible but it would take a lot of work.

> #561 seems part of this PR. Lets close it.

Yeah sorry, i was working in the master branch instead of creating a separate one -,-

-----

1, The [postcss-assets](https://github.com/assetsjs/postcss-assets) should be mentioned somewhere in the readme to avoid confusion and point out the URL resolution possibilities.

2, The [Add-custom-Gulp-task
](https://github.com/mgechev/angular2-seed/wiki/Add-custom-Gulp-task) wiki page should also be updated, to add sass and source maps. Since you only need to drop 3 `.pipe` to build.html_css.ts it should be another wiki article. And the add custom task article should be changed to add some other rudimentary task.

**Comment by mgechev** [comment]
I'm wondering if postcss-assets will have more negative rather than a positive impact on the seed.

In general I find it very useful and I'd include it in my project. On the other hand, it brings one more level of indirection and may be confusing for the developers who are not familiar with it.

// cc @ludohenin @nareshbhatia @Bigous

**Comment by tsm91** [comment]
Correct, it is certainly possible. In fact it is very handy to just write `resolve()` to font-face and background urls and it will resolve the file no matter where it is inside of `/src/assets`. 

I think it is very easy to pick it up, also this is an opt-in feature so it is not necessary to use it. A few lines of comment right above `background: resolve("more.svg");` would clear the confusion i think.

**Comment by ludohenin** [comment]
I'm not familiar with `postcss` stuffs so difficult to say. If it's an opt-in then that's probably fine.

**Comment by d3viant0ne** [comment]
Postcss takes a bit of mental remapping compared to most of the current tools. That said, it's powerful / fast and pluggable which makes it attractive. I've only tinkered with it & I have thus far been impressed.

IMO - So long as it is opt-in and well documented, I think it has a place here. It's rapidly gained in popularity and the plugin ecosystem seems to double every time I go look.

**Comment by mgechev** [comment]
@tsm91 a rebase will be required. There are a lot of changes because of the latest refactoring which aims bigger modularity of the seed so resolving the merge conflicts will not be trivial...

If you want you can open a new PR. about `postcss-assets`, how about not including them in the build process but adding a sample integration page in the wiki? How does it sound?

**Comment by d3viant0ne** [comment]
@tsm91 @mgechev - I agree with the sample integration page idea for `postcss-assets` keeps things nice and pre/post processor agnostic.

@tsm91 - As a side note, having just done this with a pre-refactoring branch. It was significantly faster to create a new branch and recommit the changes than it was to try and work them into a rebase.

**Comment by mgechev** [comment]
@tsm91 any updates?

**Comment by tsm91** [comment]
@d3viant0ne @mgechev hey guys, sorry i am in a middle of a moving process

had to paint the walls of the flat, package up my stuff, but i will do this today and re-submit the PR

i will close this one and open a new one, the re-base process seem to be scary :D

**Comment by mgechev** [comment]
No worries. There were some huge changes in the seed so I agree that a new PR will be an easier option. Looking forward for it! Btw, lets not include `postcss-assets` initially. I'll be happy if we have a wiki article for it (since I'll be one of the users of it) but the level of indirection it brings could be a bit frustrating.

**Comment by tsm91** [comment]
@mgechev roger that, i wont include `postcss-assets` only as a wiki tutorial

**Comment by mgechev** [comment]
@tsm91 sounds awesome! Looking forward for it!

**Comment by sullyme** [comment]
`css-lint` disappeared?

**Pull request closed without merging**